File: hackerrank/30 Days of Code/Day 3/Conditionals.scala
object Conditionals {

  def main(args: Array[String]) {
    val stdin = scala.io.StdIn

    val N = stdin.readLine.trim.toInt
    var res: String = ""
    if (N % 2 == 1) {
      res = "Weird"
    } else {
      if (N < 6) {
        res = "Not Weird"
      } else if (N <= 20) {
        res = "Weird"
      } else {
        res = "Not Weird"
      }
    }

    println(res)
  }
}

File: hackerrank/30 Days of Code/Day 3/conditionals.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}



function main() {
    const N = parseInt(readLine(), 10);
    let statement = "";
    if (N % 2 == 0) {
        if (N <= 5) {
            statement = "Not Weird";
        } else if (N <= 20) {
            statement = "Weird";
        } else {
            statement = "Not Weird";
        }
    } else {
        statement = "Weird";
    }
    process.stdout.write(statement);
}
